pkg/init/backend/interactive.go

```diff
@@ -104,3 +106,149 @@ func (o *InteractiveBackend) PersonalizeName(devfile parser.DevfileObj, flags ma
 	}
 	return devfile.SetMetadataName(name)
 }
+
+func (o *InteractiveBackend) PersonalizeDevfileconfig(devfileobj parser.DevfileObj) error {
```

Comment:
As discussed with Tomas on Slack, could you pass the devfile data only here, and execute `devfileobj.WriteYamlDevfile()` at the CLI level, so we remove the Filesystem dependency for this method?

Reply:
I feel it makes more sense to  keep things as is. There is `PersonalizeName` as well that has filesystem dependency, to be consistent with it, I think current code should work just fine.